intake_dal/dal_online.py

```diff
@@ -0,0 +1,188 @@
+from urllib.parse import urlparse, ParseResult, urldefrag  # noqa: F401
+import base64
+import io
+import json
+import urllib.parse
+from collections import Iterable
+from typing import Dict, Optional
+
+import numpy as np
+import pandas as pd
+import pandavro
+import pkg_resources
+import requests
+from intake import DataSource, Schema
+
+
+class DalOnlineSource(DataSource):
+    """
+    Interfaces with Online FS
+    This Intake source is to be used in conjunction with intake-dal
+    """
+    container = "dataframe"
+    partition_access = False
+    name = "dal-online"
+    version = pkg_resources.get_distribution("intake-dal").version
+
+    def __init__(self, urlpath="", key=None, storage_options=None, metadata=None):
+        """
+            Fetches from Online FS
+        Args:
+            urlpath: the host path
+            key: key value
+            key_name: key name
+            storage_options: optional storage options to send along
+            metadata:
+        """
+
+        self._urlpath = urlpath
+        parse_result = urlparse(urlpath)  # type: ParseResult
+        (self._url, _) = urldefrag(urlpath)
+        self._key_name = parse_result.fragment
+        assert self._key_name != '', "key_name expected in URL fragment"
+        self._key_value = key
+        self._canonical_name = None  # _get_schema() sets this
+        super(DalOnlineSource, self).__init__(metadata=metadata)
+
+    def _get_schema(self):
+        if self._canonical_name is None:
```

Comment:
so this method needs to get called in every `write()`? Can it be instantiated once (once metadata is set by parent class I guess?)

Reply:
it only runs if `self._canonical_name is None`